src/utils.ts

```diff
@@ -266,6 +274,176 @@ export const getConfigFilesChangedInPR = async (pr: devOps.PullRequestProperties
   }
 };
 
+
+/**
+ * 
+ * @param pr 
+ */
+export const getChangedFilesFromPR = async (
+  pr: devOps.PullRequestProperties | undefined
+): Promise<string[]> => {
+  if (pr) {
+    const filesChanged = (await pr.diff()).map((file) => file.path);
+    return filesChanged;
+  }
+  return [];
+}; 
+
+/**
+ * get all tags who is containing the changed files
+ * @returns {array} ["tag1","tag2"]
+ * @param markDownEx 
+ * @param specsChanged 
+ */
+const getTagsForFilesChanged = (
+        markDownEx: MarkDownEx,
+        specsChanged: readonly string[]
+    ): readonly string[] => {
+        const codeBlocks = getTagsToSettingsMapping(markDownEx.markDown);
+        const tagsAffected = new Set<string>();
+
+        for (const [tag, settings] of sm.entries(codeBlocks)) {
+            // for every file in settings object, see if it matches one of the
+            // paths changed
+            const filesTouchedInTag = specsChanged.filter(
+                spec => inputFile(settings).some(inputFile => spec.includes(inputFile))
+            )
+
+            if (filesTouchedInTag.length > 0) {
+                tagsAffected.add(tag)
+            }
+        }
+        return [...tagsAffected]
+    }
+
+
+/**
+ * Gets the path to the readme starting with "specification" folder and without returning the file name
+ * @param readmeUrl Full Url path to readme
+ */
+const getReadMeRelativePathToRepoWithoutFileName = (readmeUrl: string): string => {
```

Comment:
The name is confusing. Does it means getReadMeRelativeDirPath?

Reply:
fixed